vscode-dotty/src/extension.ts

```diff
@@ -16,73 +16,132 @@ export function activate(context: ExtensionContext) {
   extensionContext = context
   outputChannel = vscode.window.createOutputChannel('Dotty Language Client');
 
-  const artifactFile = `${vscode.workspace.rootPath}/.dotty-ide-artifact`
-  fs.readFile(artifactFile, (err, data) => {
-    if (err) {
-      outputChannel.append(`Unable to parse ${artifactFile}`)
-      throw err
-    }
-    const artifact = data.toString().trim()
-
-    if (process.env['DLS_DEV_MODE']) {
-      const portFile = `${vscode.workspace.rootPath}/.dotty-ide-dev-port`
-      fs.readFile(portFile, (err, port) => {
-        if (err) {
-          outputChannel.append(`Unable to parse ${portFile}`)
-          throw err
+  const sbtArtifact = "org.scala-sbt:sbt-launch:1.2.0"
+  const buildSbtFile = `${vscode.workspace.rootPath}/build.sbt`
+  const dottyPluginSbtFile = path.join(extensionContext.extensionPath, './out/dotty-plugin.sbt')
+  const languageServerArtifactFile = `${vscode.workspace.rootPath}/.dotty-ide-artifact`
+  const languageServerDefaultConfigFile = path.join(extensionContext.extensionPath, './out/default-dotty-ide-config')
+  const coursierPath = path.join(extensionContext.extensionPath, './out/coursier');
+
+  if (process.env['DLS_DEV_MODE']) {
+    const portFile = `${vscode.workspace.rootPath}/.dotty-ide-dev-port`
+    fs.readFile(portFile, (err, port) => {
+      if (err) {
+        outputChannel.append(`Unable to parse ${portFile}`)
+        throw err
+      }
+
+      run({
+        module: context.asAbsolutePath('out/src/passthrough-server.js'),
+        args: [ port.toString() ]
+      })
+    })
+
+  } else {
+    // Check whether `.dotty-ide-artifact` exists. If it does, start the language server,
+    // otherwise, try propose to start it if there's no build.sbt
+    if (fs.existsSync(languageServerArtifactFile)) {
+      runLanguageServer(coursierPath, languageServerArtifactFile)
+    } else if (!fs.existsSync(buildSbtFile)) {
+      vscode.window.showInformationMessage(
+          "This looks like an unconfigured project. Would you like to start Dotty IDE?",
```

Comment:
"an unconfigured Scala project", "start the Dotty IDE"

Also maybe remember the choice of the user by creating an empty file ".dotty-ide-disabled" ?